src/tests/NoSqlProviderTests.ts

```diff
@@ -2052,6 +2055,107 @@ describe('NoSqlProvider', function () {
+                            return openProvider(provName, {
+                                version: 1,
+                                stores: [
+                                    {
+                                        name: 'test',
+                                        primaryKeyPath: 'id',
+                                        indexes: [
+                                            {
+                                                name: 'ind1',
+                                                keyPath: 'content',
+                                            }
+                                        ]
+                                    }
+                                ]
+                            }, true).then(prov => {
+                                return prov.put('test', { id: 'abc', content: 'ghi' }).then(() => {
+                                    return prov.close();
+                                });
+                            }).then(() => {
+                            const storeSpy = sinon.spy(SqlTransaction.prototype, 'getStore');
+                            return openProvider(provName, {
+                                    version: 2,
+                                    stores: [
+                                        {
+                                            name: 'test',
+                                            primaryKeyPath: 'id'
+                                        }
+                                    ]
+                                }, false).then(prov => {
+                                    // NOTE - the line below tests an implementation detail
+                                    sinon.assert.notCalled(storeSpy);
+
+                                    // check the index was actually removed
+                                    const p1 = prov.get('test', 'abc').then((item: any) => {
+                                        assert.ok(item);
+                                        assert.equal(item.id, 'abc');
+                                        assert.equal(item.content, 'ghi');
+                                    });
+                                    const p2 = prov.getOnly('test', 'ind1', 'ghi').then((items: any[]) => {
+                                        assert.equal(items.length, 1);
+                                        assert.equal(items[0].id, 'abc');
+                                    }).then(() => {
+                                        assert.ok(false, 'should not work');
+                                    }, err => {
+                                        return SyncTasks.Resolved();
+                                    });
+                                    return SyncTasks.all([p1, p2]).then(() => {
+                                        return prov.close();
+                                    }).always(() => {
+                                        storeSpy.restore();
```

Comment:
What if the promises hang and the test times out? This `storeSpy.restore()` will not be called.